gms/gossiper.cc

```diff
@@ -2311,6 +2311,9 @@ future<> gossiper::wait_alive(std::vector<gms::inet_address> nodes, std::chrono:
 }
 
 future<> gossiper::wait_for_live_nodes_to_show_up(size_t n) {
+    if (this_shard_id() != 0) {
+        on_internal_error(logger, "Must be called on shard 0");
+    }
     logger::rate_limit rate_limit{std::chrono::seconds{5}};
 #ifdef SEASTAR_DEBUG
     // Account for debug slowness. 3 minutes is probably overkill but we don't want flaky tests.
```

Comment:
Nitpick: This comments talks about "3 minutes" but below you also add another 6 minute timeout. Would have been nice to have a comment which clarifies what all these numbers mean together.

Reply:
okay